src/update-feeds/update-transactions.ts

```diff
@@ -135,6 +141,13 @@ export const deriveSponsorWallet = (sponsorWalletMnemonic: string, dapiName: str
   );
   logger.debug('Derived sponsor wallet', { sponsorAddress, sponsorWalletAddress: sponsorWallet.address });
 
+  updateState((draft) => {
+    draft.derivedSponsorWallets = {
+      ...draft.derivedSponsorWallets,
+      [dapiName]: sponsorWallet.privateKey,
+    };
+  });
```

Comment:
🕵️ could also:
```suggestion
  updateState((draft) => {
    draft.derivedSponsorWallets[dapiNake] = sponsorWallet.privateKey
  });
```

Reply:
The current implementation is a cheat because `draft.derivedSponsorWallets` may be undefined. The spread approach creates the object if it doesn't exist.

Reply:
ah right! Feel free to ignore then 😄